zio-http/src/main/scala/zhttp/service/Handler.scala

```diff
@@ -154,29 +138,10 @@ private[zhttp] final case class Handler[R](
         if (self.isWebSocket(res)) {
           self.upgradeToWebSocket(ctx, jReq, res)
         } else {
-          unsafeWriteAnyResponse(res)
-
+          unsafeWriteAndFlushAnyResponse(res)
           res.data match {
-            case HttpData.Empty =>
-              unsafeWriteAndFlushLastEmptyContent()
-              releaseRequest(jReq)
-
-            case data @ HttpData.Text(_, _) =>
-              unsafeWriteLastContent(data.encodeAndCache(res.attribute.memoize))
-              releaseRequest(jReq)
-
-            case HttpData.BinaryByteBuf(data) =>
-              unsafeWriteLastContent(data)
-              releaseRequest(jReq)
-
-            case data @ HttpData.BinaryChunk(_) =>
-              unsafeWriteLastContent(data.encodeAndCache(res.attribute.memoize))
-              releaseRequest(jReq)
-
-            case HttpData.BinaryStream(stream) =>
-              unsafeRunZIO(
-                writeStreamContent(stream) *> Task(releaseRequest(jReq)),
-              )
+            case HttpData.BinaryStream(stream) => unsafeRunZIO(writeStreamContent(stream) *> Task(releaseRequest(jReq)))
```

Comment:
This might introduce a memory leak when writeStreamContent fails `releaseRequest(jReq)` might not get called.

Reply:
Yes, you are right. I think this issue is there even in the current main.

Reply:
I will raise an separate issue for this